packages/yarnpkg-core/sources/Configuration.ts

```diff
@@ -1062,18 +1064,33 @@ export class Configuration {
     // Now that the configuration object is almost ready, we need to load all
     // the configured plugins
 
-    const plugins = new Map<string, Plugin>([
-      [`@@core`, CorePlugin],
-    ]);
-
     const getDefault = (object: any) => {
       return `default` in object ? object.default : object;
     };
 
-    if (pluginConfiguration !== null) {
+    // The order is important, otherwise it will be wrong
+    // 1. load the core plugins
+    // 2. completely load the rcfiles
+    // 3. load third-party plugins
```

Comment:
But if the rc files contain settings from third-party plugins it won't work, correct?

Reply:
You're right, I missed it 😨.

I changed `pickCoreFields`, `excludeCoreFields` to `pickPrimaryCoreFields`, `pickSecondaryCoreFields`, `pickPluginFields` to make it easier to control the flow

The current order
1. load some core fields (importSettings + useWithSource)
2. do something
3. load all the core fields (importSettings + useWithSource)
4. load official plugins (importSettings)
5. load third-party plugins (importSettings)
6. load all the plugin fields (useWithSource)

https://github.com/yarnpkg/berry/pull/4912/commits/4a8cb17f87062904ed5a0763d4f2ead314aab382

new test for load the plugin configuration:

https://github.com/yarnpkg/berry/pull/4912/commits/cc33cb1b8d2fcd57fedc130c8bf4f2e2a64cadf3

Reply:
Oh no, my test is broken on windows, I'll fix it😥

Reply:
Done~